### PR TITLE
KYLIN-4122 Add kylin user and group manage modules

### PR DESCRIPTION
**Pull request by luguosheng1314**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java [hunk 1]
```diff
@@ -46,6 +78,28 @@
...
+
+    @Autowired
+    @Qualifier("accessService")
+    private AccessService accessService;
+
+    @Autowired
+    @Qualifier("userGroupService")
+    private UserGroupService userGroupService;
+
+    private Pattern passwordPattern;
```
Comment (nichunen) [inline server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java]: Inappropriate order for these variables

File: server-base/src/main/java/org/apache/kylin/rest/service/KylinUserService.java [hunk 2]
```diff
@@ -130,6 +133,16 @@ public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundEx
         return getKylinUserManager().list();
     }
 
+    @Override
+    public List<ManagedUser> listUsers(String userName, Boolean isFuzzMatch) throws IOException {
+        List<ManagedUser> userList =  getKylinUserManager().list();
+        return getManagedUsersByFuzzMatching(userName, isFuzzMatch, userList, null);
+    }
+    @Override
```
Comment (nichunen) [inline server-base/src/main/java/org/apache/kylin/rest/service/KylinUserService.java]: Blank line should be inserted between methods

File: server-base/src/main/java/org/apache/kylin/rest/util/AclPermissionUtil.java [hunk 3]
```diff
@@ -55,4 +55,17 @@ private AclPermissionUtil() {
         }
         return groups;
     }
+    public static boolean isAdmin() {
```
Comment (nichunen) [inline server-base/src/main/java/org/apache/kylin/rest/util/AclPermissionUtil.java]: looks duplicate with BasicController.isAdmin

**Pull request merged**